src/SwapRouter.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {IUniversalRouter} from "src/interfaces/IUniversalRouter.sol";
+import {ICreatorToken} from "src/interfaces/ICreatorToken.sol";
+import {IERC20} from "openzeppelin/token/ERC20/IERC20.sol";
+
+contract SwapRouter {
+  IUniversalRouter private constant UNIVERSAL_ROUTER =
+    IUniversalRouter(0x198EF79F1F515F02dFE9e3115eD9fC07183f02fC); // Base
+
+  address public constant WETH_ADDRESS = 0x4200000000000000000000000000000000000006;
+  address public constant USDC_ADDRESS = 0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913;
+  bytes private path;
+  // for WRAP_ETH, V3_SWAP_EXACT_OUT, and UNWRAP_WETH
+  bytes private constant COMMANDS =
+    abi.encodePacked(bytes1(uint8(0x0b)), bytes1(uint8(0x01)), bytes1(uint8(0x0c)));
+  bytes3 private constant LOW_FEE_TIER = bytes3(uint24(500));
+
+  constructor() {
+    // WETH -> USDC, order is switched because of V3_SWAP_EXACT_OUT
+    path =
+      bytes.concat(bytes20(address(USDC_ADDRESS)), LOW_FEE_TIER, bytes20(address(WETH_ADDRESS)));
+  }
+
+  // You can get a quote for the amount of ETH you have to pay to buy a creator token by calling
+  // Uniswap's `QuoterV2` contract off-chain.
+  // Check Uniswap docs: https://docs.uniswap.org/contracts/v3/reference/periphery/lens/QuoterV2 and
+  // `quote` function in `test/SwapRouter.fork.8453.t.sol.sol`
+  function buyWithEth(address _creatorToken, uint256 _maxPayment) external payable {
+    buyWithEth(_creatorToken, msg.sender, _maxPayment);
+  }
+
+  function buyWithEth(address _creatorToken, address _to, uint256 _maxPayment) public payable {
+    _swapEthForUSDC(_creatorToken, msg.sender, 1, msg.value);
+    IERC20(USDC_ADDRESS).approve(address(_creatorToken), type(uint256).max);
```

Comment:
A couple of thoughts:

* we can probably pull this in to an internal helper method like `_approveCreatorToken(address)` since we call it in several places
* That method can also do check to see if the approval is already greater than the max payment and skip the external call if so—it's an optimization but it's so easy to do I don't see a downside. This will save gas for all calls to this creator token after the first.